tests/suites/casper/stop.js

```diff
@@ -0,0 +1,165 @@
+        test.fail("This test should not be executed.");
+    });    
+    casper.run(function() {
+        test.pass("run has been stopped");
+        test.done();
+    });
+});
+
+casper.test.begin('Casper.stop() and restart a new run', 2, function(test) {
+    casper.start();
+    casper.then(function(){
+        test.fail("This test should not be executed.");
+    });
+    casper.stop().run(function() {
+        casper.start();
+        casper.then(function(){
+            test.pass("This test should not be executed.");
+        });
+        casper.run(function() {
+            test.pass("run has been stopped");
+            test.done();
+        });
+    });
+});
+
+//----------------------------------------------------------------------
+casper.test.begin('Casper.abort() can stop before to run test', 1, function(test) {
+    casper.start();
+    casper.then(function(){
+        test.fail("This test should not be executed.");
+    });
+    casper.abort(function(){
+        test.pass("run has been aborted");
+        test.done();
+    });
+    casper.run(function() {
+        test.fail("run has not been aborted");
+        
+    });
+});
+
+casper.test.begin('Casper.abort() can stop a running test', 2, function(test) {
+    casper.start();
+    casper.then(function(){
+        test.pass("This test should be executed.");
+        casper.abort(function(){
+            test.pass("run has been aborted");
+            test.done();
+        });
+    });
+    casper.then(function(){
+        test.fail("This test should not be executed.");
+    });    
+    casper.run(function() {
+        test.fail("run has not been aborted");
+        test.done();
+    });
+});
+
+//
+
+casper.test.begin('Casper.abort() can stop on timeout', 1, function(test) {
+    casper.start();
+    var onWaitTimeoutFn = casper.options.onWaitTimeout;
+    var waitTimeoutVar = casper.options.waitTimeout;
+    casper.options.waitTimeout = 500;
+
+    casper.options.onWaitTimeout = function _onWaitTimeout(timeout) {
+        this.options.onWaitTimeout = onWaitTimeoutFn;
+        this.options.waitTimeout = waitTimeoutVar;    
+        casper.abort(function(){
+            test.pass("run has been aborted");
+            test.done();
+        });
+    };
+
+    casper.then(function(){
+        this.waitFor(function check() {
+            return 1 === 0;
+        },function(){});
+    });
+    casper.then(function(){
+        test.fail("This test should not be executed.");
+    });    
+    casper.run(function() {
+        test.fail("run has not been aborted");
+    });
+});
+
+casper.test.begin('Casper.abort() and restart a new run', 3, function(test) {
+    casper.start();
+    casper.then(function(){
+        test.fail("This test should not be executed.");
+    });
+    
+    casper.abort(function(){
+        test.pass("run has been aborted");
+        casper.start();
+        casper.then(function(){
+            test.pass("This test should not be executed.");
```

Comment:
Hi Natim, 
Thanks for be interested in my feature

>> Do you mean test.fail here?

It never fails because of casper.abort is called before it could happen

Why should it pass if it should not be executed?

these comment is a mistake  of copy paste , "This test should be executed" is that I planned to write.